### ISSUE TITLE
Font can't play with pkg_resources and py2exe (70)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#85]
#  **Issue &numero;70 opened by *[illume](https://github.com/illume)* at 2011-08-22 09:40:26**

Originally reported by: **René Dudfield (Bitbucket: [illume](https://bitbucket.org/illume), GitHub: [illume](https://github.com/illume))**

----------------------------------------

== Jamie Lennox, 2011-05-13 01:18:44 -0700 ==

    
    
    For whatever reason, when i freeze an application with py2exe using the default font (eg Font(None, size)) fails and it parachutes. 
    
    What i've found to get around it is to include a font and specify it by name. This works, but what i would like to do in this case is to include the font in my egg/exe and use pkg_resources to get it out. 
    
    pkg_resources.resource_stream gives a python file handle. 
    pkg_resources.resource_string gives a string buffer of the file. 
    pkg_resources.resource_filename is supposed to extract the file from the egg to a tmp dir and return a filename, but it doesn't work with zips and frozen (throws NotImplemented).
    
    I guess the major bug is why py2exe and fonts don't work together.
    The more easily achievable (maybe, i don't know how openttf works) would be to allow Font to take a filehandle such that i could just pass the resource_stream output to it.
    
    

----

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/70
<hr>

#  Comments

# # #  **[illume](https://github.com/illume)* commented at 2011-09-06 17:04:19*

*Original comment by* **René Dudfield (Bitbucket: [illume](https://bitbucket.org/illume), GitHub: [illume](https://github.com/illume))**:

----------------------------------------

We need to make it easier for py2exe to include the default font, and for Font(None) to find the default font in a py2exe situation.

<hr>

# # #  **[MyreMylar](https://github.com/MyreMylar)* commented at 2019-11-12 11:32:14*

After a bit of research I think pygame should join matplotlib in abandoning support for py2exe. 

In short py2exe has not updated with support beyond python 3.4. The matplotlib library had the same problem that we had here with py2exe not supporting modules with included data files (like pygame's default font). Given how out of data py2exe is matplotlib dropped their support for it last year (I think they were the only library that had specific support for it anyway).

There are now several alternatives for packaging python programs, the most prominent being PyInstaller.

**Relevant related links:**

https://stackoverflow.com/questions/32963057/is-there-a-py2exe-version-thats-compatible-with-python-3-5 

https://github.com/matplotlib/matplotlib/pull/12043
